pkg/pool-manager/pool_test.go

```diff
@@ -375,6 +378,66 @@ var _ = Describe("Pool", func() {
 				Expect(exist).To(BeFalse())
 			})
 		})
+		Context("check create a vm with mac address allocation", func() {
+			var (
+				newVM        *kubevirt.VirtualMachine
+				poolManager  *PoolManager
+				allocatedMac string
+			)
+			BeforeEach(func() {
+				poolManager = createPoolManager("02:00:00:00:00:00", "02:00:00:00:00:01", &vmConfigMap)
+				newVM = masqueradeVM.DeepCopy()
+				newVM.Name = "newVM"
+
+				By("Create a VM")
+				err := poolManager.AllocateVirtualMachineMac(newVM)
+				Expect(err).ToNot(HaveOccurred(), "should successfully  allocated macs")
+
+				allocatedMac = newVM.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress
+				By("check mac allocated as expected")
+				Expect(allocatedMac).To(Equal("02:00:00:00:00:01"), "should successfully allocate the first mac in the range")
+
+				By("check configmap updated with mac allocated")
+				configMap, err := poolManager.kubeClient.CoreV1().ConfigMaps(poolManager.managerNamespace).Get(context.TODO(), names.WAITING_VMS_CONFIGMAP, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred(), "should successfully get configmap")
+
+				macAddressInConfigMapFormat := strings.Replace(allocatedMac, ":", "-", 5)
+				Expect(len(configMap.Data)).To(Equal(1), "configmap should hold the mac address waiting for approval")
+				fmt.Printf("configMap Data %v\n", configMap.Data)
+				_, exist := configMap.Data[macAddressInConfigMapFormat]
+				Expect(exist).To(Equal(true), "should have an entry of the mac in the configmap")
+
+				By("check allocated pool is populated and set to AllocationStatusWaitingForPod status")
+				Expect(len(poolManager.macPoolMap)).To(Equal(1), "macPoolMap should hold the mac address waiting for approval")
+				Expect(poolManager.macPoolMap[allocatedMac]).To(Equal(AllocationStatusWaitingForPod), "macPoolMap's mac's status should be set to AllocationStatusWaitingForPod status")
+
+				By("mark the vm as allocated")
+				err = poolManager.MarkVMAsReady(newVM, log.WithName("fake-Reconcile"))
```

Comment:
Let's move `MarkVMAsReady` to a nested `Context` with BeforeEach so we can test more stuff at `It`